generic/Storage.c

```diff
@@ -47,7 +68,7 @@ static int torch_Storage_(new)(lua_State *L)
       storage = THStorage_(newWithDataAndAllocator)(ptr, size, allocator, NULL);
     else
       storage = THStorage_(newWithData)(ptr, size);
-    storage->flag = 0;
+    storage->flag = TH_STORAGE_REFCOUNTED;
```

Comment:
The original code here was wrong: even if you create a storage from an existing data buffer (externally managed), you still want to be refcounted so that the storage struct can get freed. You just don't want to have TH_STORAGE_FREEMEM.